### ISSUE TITLE
Can we make payments reversible?

### ISSUE THREAD

**Issue by ripper234** [unsystem/paypub#2]
Scenario: An attacker says they will publish the information if X BTC are gathered at an address.
X BTC are indeed collected there, but the attacker then just does not publish the information, effectively locking the bitcoins there.

Can we devise a scheme, perhaps using multisig and/or nLockTime, that will allow contributors to 'chargeback' their contribution in this scenario?

**Comment by genjix** [comment]
maybe with SIGHASH_ANYONE_CAN_PAY.

basically the valid inputs are collected, but the tx to an address only becomes valid once enough bitcoins are collected. however this would require a centralised point to collect the inputs which isn't ideal.

locktime isn't really effective for anything. not sure about how the multisig scenario could work either.

**Comment by petertodd** [comment]
We can! You can either do it interactively by having the publisher sign a nLockTime'd refund transaction, or non-interactively once my CHECKLOCKTIMEVERIFY opcode is added to Bitcoin.

The issue is that without doing one of the above the customer can watch the network for attempts by the publisher to redeem their transaction - revealing the key - and then double-spend them, thus getting the data without paying.

**Comment by genjix** [comment]
I just noticed this on the CHECKLOCKTIMEVERIFY BIP, very nice.

**Comment by petertodd** [comment]
Thanks! Bitcoin will probably get it in about a year or so; Viacoin hopefully sooner.

**Comment by ripper234** [comment]
@petertodd - do you mean the Bitcoin will incorporate CHECKLOCKTIMEVERIFY only in a year? Or did you mean something else?

**Comment by petertodd** [comment]
@ripper234 Yeah, @laanwj doesn't think it'll be ready in time for v0.10, so the earliest would then be v0.11 Including waiting for miners to upgrade that's probably about a year from now, if not even longer.

**Comment by laanwj** [comment]
@petertodd Rolling it out for 0.10 would be extremely short term, on the other hand as BIP62 requires a softfork anyhow, it may be better to do it in one go.

**Comment by petertodd** [comment]
@laanwj Yup, lots more testing to be done...

**Comment by genjix** [comment]
yep this is really cool
